### DISCUSSION TITLE
First of two designs for a MiniDexed IO module

### DISCUSSION THREAD

**Discussion by diyelectromusic** [probonopd/MiniDexed#335] [Show and tell]
I've now received, built and tested one of my pcb designs for an IO module for MiniDexed.  I quite like this one as it only needs power, audio and MIDI and then it is ready to go. I have another one to write up shortly (and I'll post again here when its done) that is a bigger board, but has all the IO off to the back and side, so might be more "mountable" e.g. in a case.

This one supports: SSD1306 128x32 OLED display, PCM5102 DAC, MIDI IN, rotary encoder and two buttons.

Note: If you are interested in following along with this one, note that I'm still quite a novice when it comes to pcb design.  Also note that the MIDI IN design borrows heavily from Clumsy MIDI (which I use with my MT-32 Pi and can highly recommend too).

My build and design notes can be found here: https://diyelectromusic.wordpress.com/2022/08/16/minidexed-raspberry-pi-io-board-part-2/

Kevin

![IMG_6385](https://user-images.githubusercontent.com/68612569/184933382-d173eb5c-bad6-4893-9e6b-2e1074b89ece.JPG)

**Comment by diyelectromusic** [comment]
And here are the notes for the slightly larger version that has all IO (apart from the PI's USB, etc) at the back.

https://diyelectromusic.wordpress.com/2022/08/16/minidexed-raspberry-pi-io-board-part-3/

This one supports the original 16x2 display and has three MIDI ports (IN, OUT, THRU), although there are options to drop that down to two.

All the same provisos re my competency at pcb design apply! :)

![IMG_6394](https://user-images.githubusercontent.com/68612569/184993791-ceffc203-748b-4bde-8d4f-61fb2c59d9b2.JPG)
![IMG_6393](https://user-images.githubusercontent.com/68612569/184993794-0229ca06-b28a-494b-b3fe-a870396eab5a.JPG)

**Reply by probonopd** [reply]
Great work @diyelectromusic. Congrats!

**Reply by diyelectromusic** [reply]
Ping me an email if you'd like one... (although I know you've already got a few nice units built of your own!)

**Reply by probonopd** [reply]
These PCBs look amazing. Can't wait to populate them :+1:

**Reply by mbwum73** [reply]
Thank you very much! I made the second version with the 3 midi ports. It works great, however I have a small issue which I can't fix. During boot the LCD displayes a bit of garbage. I think it should show "Loading" or something like that, but I see garbage. Once everything is loaded it works fine and the display is just fine. So just during boot I see some messy characters on the screen. Anyone knows what's causing this?

Also wondering if anyone made a nice 3D printed case for the version with the 3 midi connectors.

If anyone is interested, I have a few pcb's left (had to order 5 minimum).

**Reply by johndoe201274** [reply]
hi there, im very interested in the pcb. kind regards,

**Reply by johndoe201274** [reply]
im working on a 3d model....if i have the model i'll let you know...

**Comment by pinkmouse101** [comment]
Very neat work. Don't suppose you have a spare PCB? I'm currently designing a case, and I think your big design would be a perfect fit!

[Screenshot 2022-08-18 at 00 09 37](https://user-images.githubusercontent.com/111195179/185259323-5fd96f5f-ccb3-4d64-abb6-736ea4397491.png)

**Reply by diyelectromusic** [reply]
> Very neat work. Don't suppose you have a spare PCB? I'm currently designing a case, and I think your big design would be a perfect fit!
> 
> [Screenshot 2022-08-18 at 00 09 37](https://user-images.githubusercontent.com/111195179/185259323-5fd96f5f-ccb3-4d64-abb6-736ea4397491.png)

Drop me an email diyelectromusic@gmail.com and we can negotiate! :)

**Reply by pinkmouse101** [reply]
Will do! :)

**Reply by seymourgler** [reply]
> I'm currently designing a case...

@pinkmouse101 did that enclosure work out? If so, would you mind sharing the 3D print file?

**Reply by pinkmouse101** [reply]
I thought I had!

Will put it somewhere public when I get home from work

**Reply by pinkmouse101** [reply]
Oops, got confused with github projects, I ended up making a module for a PI one to fit a couple along with other stuff in a 1u rack case:
![mini](https://github.com/probonopd/MiniDexed/assets/111195179/f0118f58-9899-4979-85a4-97572ced7bf8)

I'll gladly share the files if they're any use to you.

**Comment by poetaster** [comment]
If anyone is searching, I have some spares of the version one build as pictured here:

![2448-2448-max](https://github.com/probonopd/MiniDexed/assets/1065933/ee977565-f777-4b41-8389-9685b77a51cf)

That's the 2011 pi https://diyelectromusic.wordpress.com/2023/02/28/minidexed-raspberry-pi-v1-io-board-part-2/

**Comment by mbwum73** [comment]
Okay great! Looking forward to your 3D design.
Where in the world do you live? I can sent you a pcb, but it could be expensive if you live on the other side of the planet.


________________________________
Van: johndoe201274 ***@***.***>
Verzonden: maandag 23 december 2024 12:41
Aan: probonopd/MiniDexed ***@***.***>
CC: mbwum73 ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [probonopd/MiniDexed] First of two designs for a MiniDexed IO module (Discussion #335)


im working on a 3d model....if i have the model i'll let you know...

**Comment by probonopd** [comment]
@diyelectromusic let's make a [hwconfig](https://github.com/probonopd/MiniDexed/tree/main/hwconfig) for the boards, shall we?

**Reply by diyelectromusic** [reply]
If you don't think that clutters things up, then sure - go for it :)
I should probably properly list all my boards at some point!!

Kevin

**Comment by poetaster** [comment]
Which reminds me, I still have a few of these boards if someone wants one :) I have two in use and they're really handy ;)

**Reply by BobanSpasic** [reply]
Hi @poetaster, where are you located? I would like to get one of these. I am from Vienna, Austria

**Reply by BobanSpasic** [reply]
Sorry, I saw just now it is for Raspberry Pi 1 (first generation). I don't have a Pi1

**Reply by poetaster** [reply]
> Sorry, I saw just now it is for Raspberry Pi 1 (first generation). I don't have a Pi1

No problem. They are ancient :) But I had a bunch of them, so I made a number of Kevin's boards and still have extras. Maybe he has some designs for newer pis? @diyelectromusic  did you do a design for new pis?

**Reply by diyelectromusic** [reply]
A few :) https://diyelectromusic.com/pcbs/ (see the Raspberry Pi section) and also my more general posts on MiniDexed: https://diyelectromusic.com/tag/minidexed/

I really ought to list them all out in a page somewhere... :)

Kevin
